src/utils.ts

```diff
@@ -26,3 +35,29 @@ const AsyncFunctionConstructor = (async () => {}).constructor;
  * an async function check method only consider runtime support async syntax
  */
 export const isAsyncFunction = (fn: Fn) => fn.constructor === AsyncFunctionConstructor;
+
+export const isAsyncTask = async (task: Task) => {
+  if (isAsyncFunction(task.fn)) {
+    return true;
+  }
+  try {
+    if (task.opts.beforeEach != null) {
+      await task.opts.beforeEach.call(task);
```

Comment:
hmm

kind of feels error prone. if beforeEach or afterEach fail, then we result in the fact that the Task is not async. 

imho beforeEach and afterEach have to be in try catch blocks

Reply:
good point, thank you